comp/forwarder/defaultforwarder/forwarder_health.go

```diff
@@ -142,22 +158,46 @@ func (fh *forwarderHealth) healthCheckLoop() {
 	}
 }
 
-// computeDomainsURL populates a map containing API Endpoints per API keys that belongs to the forwarderHealth struct
-func (fh *forwarderHealth) computeDomainsURL() {
+func (fh *forwarderHealth) updateAPIKey(newKey string) {
+	fh.keyMapMutex.Lock()
+	for domainURL, keyList := range fh.keysPerAPIEndpoint {
+		replaceList := make([]string, 0, len(keyList))
+		for _, oldKey := range keyList {
+			if oldKey == fh.mainAPIKey {
+				replaceList = append(replaceList, newKey)
+			} else {
+				replaceList = append(replaceList, oldKey)
+			}
+		}
+		fh.keysPerAPIEndpoint[domainURL] = replaceList
+	}
+	fh.mainAPIKey = newKey
+	fh.keyMapMutex.Unlock()
+	// check apiKey validity and update the messages
+	fh.checkValidAPIKey()
+}
+
+// computeDomainURLAPIKeyMap populates a map containing API Endpoints per API keys that belongs to the forwarderHealth struct
+func (fh *forwarderHealth) computeDomainURLAPIKeyMap() {
+	fh.keyMapMutex.Lock()
 	for domain, dr := range fh.domainResolvers {
 		if domainURLRegexp.MatchString(domain) {
 			domain = "https://api." + domainURLRegexp.FindString(domain)
 		}
 		fh.keysPerAPIEndpoint[domain] = append(fh.keysPerAPIEndpoint[domain], dr.GetAPIKeys()...)
 	}
+	fh.keyMapMutex.Unlock()
 }
 
 func (fh *forwarderHealth) setAPIKeyStatus(apiKey string, _ string, status *expvar.String) {
 	if len(apiKey) > 5 {
 		apiKey = apiKey[len(apiKey)-5:]
 	}
 	obfuscatedKey := fmt.Sprintf("API key ending with %s", apiKey)
-	if status == &apiKeyInvalid {
+	if status == &apiKeyRemove {
```

Comment:
Are we actually ever using `apiKeyRemove` in `setAPIKeyStatus` ?

Reply:
Good catch! The usage of it got removed in a previous commit, I've put it back now. It's necessary so that when the API Key is updated to use a new one, the old messages about the validity of the previous key get cleared.